crates/ruff_python_formatter/src/expression/expr_name.rs

```diff
@@ -38,9 +38,14 @@ impl NeedsParentheses for ExprName {
     fn needs_parentheses(
         &self,
         _parent: AnyNodeRef,
-        _context: &PyFormatContext,
+        context: &PyFormatContext,
     ) -> OptionalParentheses {
-        OptionalParentheses::Never
+        let text_len = context.source()[self.range].len();
+        if text_len > 4 && text_len < context.options().line_width().value() as usize {
```

Comment:
Is 4 the length of `if (`?

Reply:
This is a rather arbitrary number and a heuristic to avoid the cost of using best fitting for most literals and common names like `self`. It can result in the formatted text exceeding the line width even if parenthesizing would make it fit but IMO, I probably rather have a 4-character name exceed the line width by 2 than add parentheses and expand it vertically. 

How did I come up with the number. I exclude the constants `True`, `False` and `None` and... I assumed that they all have a length of 4. So you see, I'm not good at math nor counting :rofl:  

I'll factor this method out and add a comment of why 4... uhm 5.